### PR TITLE
Two minor improvements

### PR DESCRIPTION
**Pull request by IvanVolosyuk**

### CODE REVIEW

File: kvs/src/storage.cpp [hunk 1]
```diff
@@ -274,16 +274,14 @@ std::string Storage::Get(const char *key) {
...
   while (true) {
     ReadNext(&offset, (char*)lens, sizeof(lens));
     if (lens[0] == 255) return "";
     uint32_t end = (offset + lens[0] + page_size_ - 1) & ~(page_size_ - 1);
-    if (lens[1] < lens[0] && lens[1] == key_length) {
+    if (lens[1] < lens[0] && lens[1] == key.length()) {
       // Key length match, read key
       ReadNext(&offset, buffer_.data(), lens[1]);
-      if (strncmp(buffer_.data(), key, key_length) == 0) {
+      if (strncmp(buffer_.data(), key.data(), key.length()) == 0) {
```
Comment (zhsh) [inline kvs/src/storage.cpp:284]: strncmp("a\0b", "a\0c", 3) seems to return 0.
Probably need "memcmp" here.

File: kvs/src/bignum.cpp [hunk 2]
```diff
@@ -34,7 +34,7 @@ BigInt BigInt::bin(const std::string& str) {
     }
     res.digits.push_back(v);
   }
-  if (res.digits.back() == 0 && res.digits.size() > 1) res.digits.pop_back();
+  while (res.digits.back() == 0 && res.digits.size() > 1) res.digits.pop_back();
```
Comment (zhsh) [inline kvs/src/bignum.cpp:37]: Maybe safer to use this:
while (res.digits.size() > 1 && res.digits.back() == 0)

File: kvs/src/storage.cpp (hunk 1 above)
Comment (IvanVolosyuk) [reply kvs/src/storage.cpp]: good point

File: kvs/src/bignum.cpp (hunk 2 above)
Comment (IvanVolosyuk) [reply kvs/src/bignum.cpp]: done

**Comment by IvanVolosyuk** [comment]
Updated

**Pull request merged**